src/darwin/Framework/CHIP/MTRAsyncWorkQueue.mm

```diff
@@ -117,8 +133,14 @@ - (void)callReadyHandlerWithContext:(id)context completion:(MTRAsyncWorkCompleti
 
     // Always dispatch even if there is no readyHandler as this avoids synchronously
     // re-entering the MTRAsyncWorkQueueCode, simplifying the implementation.
+    auto uniqueID = _uniqueID;
     auto readyHandler = _readyHandler;
     dispatch_async(_queue, ^{
+        if (!retryCount) {
+            MTR_LOG_DEFAULT("MTRAsyncWorkQueue<%@> executing work item %tu", context, uniqueID);
```

Comment:
I think we should use a consistent and easily searchable format for uniqueID, for every instance of logging the uniqueID, so that all instances of it are unique the same in the logs. I would also suggest bracketing the number so that when searching for "work item 12" it won't also match "work item 123", for example.

Reply:
Also, probably a good idea to just have a plain function that takes an ID and returns this string, and have all logging sites use that instead of constructing the string everywhere.

Reply:
> I think we should use a consistent and easily searchable format for uniqueID

Hence my suggestion for "work item N" as that format.  The function idea is nice.

Reply:
To clarify what I meant by "bracketing the number" - it's something like `work item [N]` or have some symbol after the number like `work item N!` or even `{work item N}` so that a log filter can uniquely identify work items, and not include other items that have numbers beginning with the same prefix, as in my example `work item 123` matching the pattern `work item 12`.

Just a minor detail.

Reply:
I will add the bracketing and ensure it’s consistent. Pre-composing parts of a log message into a string is generally discouraged on Darwin because it makes the log storage less efficient (os_log stores a pointer to the format string and the arguments so anything preformatted becomes more argument data)